header/sync/sync_head.go

```diff
@@ -0,0 +1,167 @@
+package sync
+
+import (
+	"context"
+	"errors"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+
+	"github.com/celestiaorg/celestia-node/header"
+)
+
+// subjectiveHead returns the latest known local header that is not expired(within trusting period).
+// If the header is expired, it is retrieved from a trusted peer without validation;
+// in other words, an automatic subjective initialization is performed.
```

Comment:
I think this must be clearly communicated in the usage instruction that setting a "trusted peer", translates into trusting any header that is received from that peer.

Reply:
> translates into trusting _any_ header that is received from that peer.
@liamsi, a small correction here. Not _any_ but only `network head` during automatic subjective init. Other than that, `network head` and past headers are verified as per logic in the PR. 

Besides the code itself, this should be clearly documented on node docs and in the config. This is though orthogonal to the PR.